Rename some of the CanMakeCheckedPtrBase member functions to make their names less generic

cdumez opened the pull request:
#### 1a981cca81c6b9825b101875b4c59e9ed238425a
<pre>
Rename some of the CanMakeCheckedPtrBase member functions to make their names less generic
<a href="https://bugs.webkit.org/show_bug.cgi?id=282459">https://bugs.webkit.org/show_bug.cgi?id=282459</a>

Reviewed by Ryosuke Niwa.

Rename some of the CanMakeCheckedPtrBase member functions to make their
names less generic. This is important since their existence is checked
by the `HasCheckedPtrMethods` type trait and given that many classes
need to define their own implementation.

* Source/WTF/wtf/CheckedPtr.h:
(WTF::CheckedPtr::refIfNotNull):
(WTF::CheckedPtr::derefIfNotNull):
* Source/WTF/wtf/CheckedRef.h:
(WTF::CheckedRef::~CheckedRef):
(WTF::CheckedRef::CheckedRef):
(WTF::CheckedRef::ptr const):
(WTF::CanMakeCheckedPtrBase::checkedPtrCount const):
(WTF::CanMakeCheckedPtrBase::incrementCheckedPtrCount const):
(WTF::CanMakeCheckedPtrBase::decrementCheckedPtrCount const):
(WTF::CanMakeCheckedPtrBase::checkedPtrCountWithoutThreadCheck const):
(WTF::CanMakeCheckedPtrBase::ptrCount const): Deleted.
(WTF::CanMakeCheckedPtrBase::incrementPtrCount const): Deleted.
(WTF::CanMakeCheckedPtrBase::decrementPtrCount const): Deleted.
(WTF::CanMakeCheckedPtrBase::ptrCountWithoutThreadCheck const): Deleted.
* Source/WTF/wtf/FastMalloc.h:
* Source/WTF/wtf/TypeTraits.h:
* Source/WebCore/Modules/badge/WorkerBadgeProxy.h:
* Source/WebCore/Modules/speech/SpeechRecognitionCaptureSourceImpl.h:
* Source/WebCore/Modules/webaudio/AudioWorkletMessagingProxy.h:
(WebCore::AudioWorkletMessagingProxy::checkedPtrCount const):
(WebCore::AudioWorkletMessagingProxy::checkedPtrCountWithoutThreadCheck const):
(WebCore::AudioWorkletMessagingProxy::incrementCheckedPtrCount const):
(WebCore::AudioWorkletMessagingProxy::decrementCheckedPtrCount const):
(WebCore::AudioWorkletMessagingProxy::ptrCount const): Deleted.
(WebCore::AudioWorkletMessagingProxy::ptrCountWithoutThreadCheck const): Deleted.
(WebCore::AudioWorkletMessagingProxy::incrementPtrCount const): Deleted.
(WebCore::AudioWorkletMessagingProxy::decrementPtrCount const): Deleted.
* Source/WebCore/dom/PendingScriptClient.h:
* Source/WebCore/dom/ScriptRunner.h:
* Source/WebCore/html/parser/HTMLDocumentParser.h:
* Source/WebCore/platform/PopupMenuClient.h:
* Source/WebCore/platform/ScrollView.h:
* Source/WebCore/platform/ScrollableArea.h:
* Source/WebCore/platform/cocoa/PlaybackSessionModel.h:
* Source/WebCore/platform/cocoa/PlaybackSessionModelMediaElement.h:
* Source/WebCore/platform/cocoa/VideoPresentationModel.h:
* Source/WebCore/platform/cocoa/WebAVPlayerLayer.mm:
* Source/WebCore/platform/graphics/cocoa/NullPlaybackSessionInterface.h:
* Source/WebCore/platform/graphics/cocoa/NullVideoPresentationInterface.h:
* Source/WebCore/platform/ios/PlaybackSessionInterfaceIOS.h:
* Source/WebCore/platform/ios/PlaybackSessionInterfaceIOS.mm:
(WebCore::PlaybackSessionInterfaceIOS::checkedPtrCount const):
(WebCore::PlaybackSessionInterfaceIOS::checkedPtrCountWithoutThreadCheck const):
(WebCore::PlaybackSessionInterfaceIOS::incrementCheckedPtrCount const):
(WebCore::PlaybackSessionInterfaceIOS::decrementCheckedPtrCount const):
(WebCore::PlaybackSessionInterfaceIOS::ptrCount const): Deleted.
(WebCore::PlaybackSessionInterfaceIOS::ptrCountWithoutThreadCheck const): Deleted.
(WebCore::PlaybackSessionInterfaceIOS::incrementPtrCount const): Deleted.
(WebCore::PlaybackSessionInterfaceIOS::decrementPtrCount const): Deleted.
* Source/WebCore/platform/ios/VideoPresentationInterfaceIOS.h:
* Source/WebCore/platform/ios/WebVideoFullscreenControllerAVKit.mm:
* Source/WebCore/platform/mac/PlaybackSessionInterfaceMac.h:
* Source/WebCore/platform/mac/PlaybackSessionInterfaceMac.mm:
(WebCore::PlaybackSessionInterfaceMac::checkedPtrCount const):
(WebCore::PlaybackSessionInterfaceMac::checkedPtrCountWithoutThreadCheck const):
(WebCore::PlaybackSessionInterfaceMac::incrementCheckedPtrCount const):
(WebCore::PlaybackSessionInterfaceMac::decrementCheckedPtrCount const):
(WebCore::PlaybackSessionInterfaceMac::ptrCount const): Deleted.
(WebCore::PlaybackSessionInterfaceMac::ptrCountWithoutThreadCheck const): Deleted.
(WebCore::PlaybackSessionInterfaceMac::incrementPtrCount const): Deleted.
(WebCore::PlaybackSessionInterfaceMac::decrementPtrCount const): Deleted.
* Source/WebCore/platform/mac/VideoPresentationInterfaceMac.h:
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivate.h:
* Source/WebCore/platform/mediastream/AudioTrackPrivateMediaStream.h:
* Source/WebCore/platform/mediastream/RealtimeMediaSource.h:
* Source/WebCore/platform/mediastream/gstreamer/GStreamerMediaStreamSource.cpp:
* Source/WebCore/platform/mediastream/libwebrtc/gstreamer/RealtimeOutgoingAudioSourceLibWebRTC.h:
* Source/WebCore/platform/mediastream/mac/MediaStreamTrackAudioSourceProviderCocoa.h:
* Source/WebCore/platform/mediastream/mac/RealtimeOutgoingAudioSourceCocoa.h:
* Source/WebCore/platform/network/curl/CurlMultipartHandleClient.h:
* Source/WebCore/platform/network/curl/CurlRequest.h:
* Source/WebCore/rendering/RenderLayerScrollableArea.h:
* Source/WebCore/rendering/RenderListBox.h:
* Source/WebCore/rendering/RenderMenuList.h:
* Source/WebCore/rendering/RenderSearchField.h:
* Source/WebCore/testing/Internals.h:
* Source/WebCore/workers/WorkerDebuggerProxy.h:
* Source/WebCore/workers/WorkerLoaderProxy.h:
* Source/WebCore/workers/WorkerMessagingProxy.h:
* Source/WebCore/workers/WorkerReportingProxy.h:
* Source/WebCore/workers/service/context/ServiceWorkerThread.cpp:
* Source/WebCore/workers/service/context/ServiceWorkerThreadProxy.h:
* Source/WebCore/workers/shared/context/SharedWorkerThreadProxy.h:
* Source/WebCore/xml/parser/XMLDocumentParser.h:
* Source/WebKit/NetworkProcess/Downloads/DownloadManager.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/UIProcess/AuxiliaryProcessProxy.h:
* Source/WebKit/UIProcess/Cocoa/PlaybackSessionManagerProxy.h:
* Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.cpp:
* Source/WebKit/UIProcess/Launcher/ProcessLauncher.h:
* Source/WebKit/UIProcess/WebPageProxyInternals.h:
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
* Source/WebKit/UIProcess/win/WebPopupMenuProxyWin.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
* Source/WebKit/WebProcess/Speech/SpeechRecognitionRealtimeMediaSourceManager.cpp:
* Source/WebKit/WebProcess/cocoa/PlaybackSessionManager.h:
* Source/WebKit/WebProcess/cocoa/VideoPresentationManager.h:
* Tools/TestWebKitAPI/Tests/WTF/CheckedPtr.cpp:
(TestWebKitAPI::TEST(WTF_CheckedPtr, Basic)):
(TestWebKitAPI::TEST(WTF_CheckedPtr, CheckedRef)):
(TestWebKitAPI::TEST(WTF_CheckedPtr, DerivedClass)):
(TestWebKitAPI::TEST(WTF_CheckedPtr, HashSet)):
(TestWebKitAPI::TEST(WTF_CheckedPtr, ReferenceCountLimit)):
* Tools/TestWebKitAPI/Tests/WTF/CheckedRef.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/curl/CurlMultipartHandleTests.cpp:

Canonical link: <a href="https://commits.webkit.org/286041@main">https://commits.webkit.org/286041@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c0de24a7fc72446fda2fb62286ab30d296ae66d5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/74553 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/53982 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/27364 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/78961 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/25791 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/76670 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/63115 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/1767 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/58598 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/16891 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/77620 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/48740 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/64100 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/38997 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/45810 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/21596 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/24124 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/67690 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/67151 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/21942 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/80497 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/73811 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/1870 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/1104 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/66871 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/2018 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/64118 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/66159 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/10086 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/8239 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/95592 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11515 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/1835 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/4622 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/20986 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/1863 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/1933 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/1870 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WTF/wtf/CheckedRef.h:
```diff
@@ -278,23 +278,23 @@ template <typename StorageType, typename PtrCounterType, DefaultedOperatorEqual
-    void incrementPtrCount() const { ++m_count; }
-    ALWAYS_INLINE void decrementPtrCount() const
+    PtrCounterType checkedPtrCount() const { return m_checkedPtrCount; }
+    void incrementCheckedPtrCount() const { ++m_checkedPtrCount; }
```
Hm... this will require clang side change.

darinadler replied:
How will we deal with this?

cdumez replied:
Oops. I had missed this comment. I have ping'd Ryosuke on Slack to see if I need to revert or not.

cdumez committed:
Rename some of the CanMakeCheckedPtrBase member functions to make their names less generic
https://bugs.webkit.org/show_bug.cgi?id=282459

Reviewed by Ryosuke Niwa.

Rename some of the CanMakeCheckedPtrBase member functions to make their
names less generic. This is important since their existence is checked
by the `HasCheckedPtrMethods` type trait and given that many classes
need to define their own implementation.

* Source/WTF/wtf/CheckedPtr.h:
(WTF::CheckedPtr::refIfNotNull):
(WTF::CheckedPtr::derefIfNotNull):
* Source/WTF/wtf/CheckedRef.h:
(WTF::CheckedRef::~CheckedRef):
(WTF::CheckedRef::CheckedRef):
(WTF::CheckedRef::ptr const):
(WTF::CanMakeCheckedPtrBase::checkedPtrCount const):
(WTF::CanMakeCheckedPtrBase::incrementCheckedPtrCount const):
(WTF::CanMakeCheckedPtrBase::decrementCheckedPtrCount const):
(WTF::CanMakeCheckedPtrBase::checkedPtrCountWithoutThreadCheck const):
(WTF::CanMakeCheckedPtrBase::ptrCount const): Deleted.
(WTF::CanMakeCheckedPtrBase::incrementPtrCount const): Deleted.
(WTF::CanMakeCheckedPtrBase::decrementPtrCount const): Deleted.
(WTF::CanMakeCheckedPtrBase::ptrCountWithoutThreadCheck const): Deleted.
* Source/WTF/wtf/FastMalloc.h:
* Source/WTF/wtf/TypeTraits.h:
* Source/WebCore/Modules/badge/WorkerBadgeProxy.h:
* Source/WebCore/Modules/speech/SpeechRecognitionCaptureSourceImpl.h:
* Source/WebCore/Modules/webaudio/AudioWorkletMessagingProxy.h:
(WebCore::AudioWorkletMessagingProxy::checkedPtrCount const):
(WebCore::AudioWorkletMessagingProxy::checkedPtrCountWithoutThreadCheck const):
(WebCore::AudioWorkletMessagingProxy::incrementCheckedPtrCount const):
(WebCore::AudioWorkletMessagingProxy::decrementCheckedPtrCount const):
(WebCore::AudioWorkletMessagingProxy::ptrCount const): Deleted.
(WebCore::AudioWorkletMessagingProxy::ptrCountWithoutThreadCheck const): Deleted.
(WebCore::AudioWorkletMessagingProxy::incrementPtrCount const): Deleted.
(WebCore::AudioWorkletMessagingProxy::decrementPtrCount const): Deleted.
* Source/WebCore/dom/PendingScriptClient.h:
* Source/WebCore/dom/ScriptRunner.h:
* Source/WebCore/html/parser/HTMLDocumentParser.h:
* Source/WebCore/platform/PopupMenuClient.h:
* Source/WebCore/platform/ScrollView.h:
* Source/WebCore/platform/ScrollableArea.h:
* Source/WebCore/platform/cocoa/PlaybackSessionModel.h:
* Source/WebCore/platform/cocoa/PlaybackSessionModelMediaElement.h:
* Source/WebCore/platform/cocoa/VideoPresentationModel.h:
* Source/WebCore/platform/cocoa/WebAVPlayerLayer.mm:
* Source/WebCore/platform/graphics/cocoa/NullPlaybackSessionInterface.h:
* Source/WebCore/platform/graphics/cocoa/NullVideoPresentationInterface.h:
* Source/WebCore/platform/ios/PlaybackSessionInterfaceIOS.h:
* Source/WebCore/platform/ios/PlaybackSessionInterfaceIOS.mm:
(WebCore::PlaybackSessionInterfaceIOS::checkedPtrCount const):
(WebCore::PlaybackSessionInterfaceIOS::checkedPtrCountWithoutThreadCheck const):
(WebCore::PlaybackSessionInterfaceIOS::incrementCheckedPtrCount const):
(WebCore::PlaybackSessionInterfaceIOS::decrementCheckedPtrCount const):
(WebCore::PlaybackSessionInterfaceIOS::ptrCount const): Deleted.
(WebCore::PlaybackSessionInterfaceIOS::ptrCountWithoutThreadCheck const): Deleted.
(WebCore::PlaybackSessionInterfaceIOS::incrementPtrCount const): Deleted.
(WebCore::PlaybackSessionInterfaceIOS::decrementPtrCount const): Deleted.
* Source/WebCore/platform/ios/VideoPresentationInterfaceIOS.h:
* Source/WebCore/platform/ios/WebVideoFullscreenControllerAVKit.mm:
* Source/WebCore/platform/mac/PlaybackSessionInterfaceMac.h:
* Source/WebCore/platform/mac/PlaybackSessionInterfaceMac.mm:
(WebCore::PlaybackSessionInterfaceMac::checkedPtrCount const):
(WebCore::PlaybackSessionInterfaceMac::checkedPtrCountWithoutThreadCheck const):
(WebCore::PlaybackSessionInterfaceMac::incrementCheckedPtrCount const):
(WebCore::PlaybackSessionInterfaceMac::decrementCheckedPtrCount const):
(WebCore::PlaybackSessionInterfaceMac::ptrCount const): Deleted.
(WebCore::PlaybackSessionInterfaceMac::ptrCountWithoutThreadCheck const): Deleted.
(WebCore::PlaybackSessionInterfaceMac::incrementPtrCount const): Deleted.
(WebCore::PlaybackSessionInterfaceMac::decrementPtrCount const): Deleted.
* Source/WebCore/platform/mac/VideoPresentationInterfaceMac.h:
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivate.h:
* Source/WebCore/platform/mediastream/AudioTrackPrivateMediaStream.h:
* Source/WebCore/platform/mediastream/RealtimeMediaSource.h:
* Source/WebCore/platform/mediastream/gstreamer/GStreamerMediaStreamSource.cpp:
* Source/WebCore/platform/mediastream/libwebrtc/gstreamer/RealtimeOutgoingAudioSourceLibWebRTC.h:
* Source/WebCore/platform/mediastream/mac/MediaStreamTrackAudioSourceProviderCocoa.h:
* Source/WebCore/platform/mediastream/mac/RealtimeOutgoingAudioSourceCocoa.h:
* Source/WebCore/platform/network/curl/CurlMultipartHandleClient.h:
* Source/WebCore/platform/network/curl/CurlRequest.h:
* Source/WebCore/rendering/RenderLayerScrollableArea.h:
* Source/WebCore/rendering/RenderListBox.h:
* Source/WebCore/rendering/RenderMenuList.h:
* Source/WebCore/rendering/RenderSearchField.h:
* Source/WebCore/testing/Internals.h:
* Source/WebCore/workers/WorkerDebuggerProxy.h:
* Source/WebCore/workers/WorkerLoaderProxy.h:
* Source/WebCore/workers/WorkerMessagingProxy.h:
* Source/WebCore/workers/WorkerReportingProxy.h:
* Source/WebCore/workers/service/context/ServiceWorkerThread.cpp:
* Source/WebCore/workers/service/context/ServiceWorkerThreadProxy.h:
* Source/WebCore/workers/shared/context/SharedWorkerThreadProxy.h:
* Source/WebCore/xml/parser/XMLDocumentParser.h:
* Source/WebKit/NetworkProcess/Downloads/DownloadManager.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/UIProcess/AuxiliaryProcessProxy.h:
* Source/WebKit/UIProcess/Cocoa/PlaybackSessionManagerProxy.h:
* Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.cpp:
* Source/WebKit/UIProcess/Launcher/ProcessLauncher.h:
* Source/WebKit/UIProcess/WebPageProxyInternals.h:
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
* Source/WebKit/UIProcess/win/WebPopupMenuProxyWin.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
* Source/WebKit/WebProcess/Speech/SpeechRecognitionRealtimeMediaSourceManager.cpp:
* Source/WebKit/WebProcess/cocoa/PlaybackSessionManager.h:
* Source/WebKit/WebProcess/cocoa/VideoPresentationManager.h:
* Tools/TestWebKitAPI/Tests/WTF/CheckedPtr.cpp:
(TestWebKitAPI::TEST(WTF_CheckedPtr, Basic)):
(TestWebKitAPI::TEST(WTF_CheckedPtr, CheckedRef)):
(TestWebKitAPI::TEST(WTF_CheckedPtr, DerivedClass)):
(TestWebKitAPI::TEST(WTF_CheckedPtr, HashSet)):
(TestWebKitAPI::TEST(WTF_CheckedPtr, ReferenceCountLimit)):
* Tools/TestWebKitAPI/Tests/WTF/CheckedRef.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/curl/CurlMultipartHandleTests.cpp:

Canonical link: https://commits.webkit.org/286041@main
